spring-cloud-dataflow-server/src/test/java/org/springframework/cloud/dataflow/integration/test/DataFlowIT.java

```diff
@@ -753,6 +756,52 @@ public void namedChannelDirectedGraph() {
 		}
 	}
 
+	@Test
+	public void dataflowTaskLauncherSink() {
+		logger.info("dataflow-tasklauncher-sink-test");
+
+		if (!runtimeApps.isAppRegistered("dataflow-tasklauncher", ApplicationType.sink)) {
+			logger.info("dataflow-tasklauncher-sink-test: SKIP - no dataflow-tasklauncher app registered!");
+		}
+
+		Assumptions.assumeTrue(runtimeApps.isAppRegistered("dataflow-tasklauncher", ApplicationType.sink),
+				"dataflow-tasklauncher-sink-test: SKIP - no dataflow-tasklauncher app registered!");
+
```

Comment:
Why do we need 2 checks here?

Reply:
The `Assumptions.assumeTrue` is the real condition that will skip the test if the apps is not registered. But could not figure out how to make it to print its message in the logs. So, the preceding  `if/then->log` block is just to log the same message for skip condition. It is important to log this message to distinct successful test run from the case when it was skipped.
It will be nice if we can avoid this repetition (used in few other tests as well), but I still don't know how.